src/sgp/Random.cc

```diff
@@ -1,69 +1,89 @@
 #include "Random.h"
-#include <stdlib.h>
-#include <time.h>
 
+#include <chrono>
+#include <random>
+
+/// Pseudo-random number engine.
+static std::mt19937 gRandomEngine;
+
+/// Uniform distribution in the range [0,UINT32_MAX].
+static std::uniform_int_distribution<UINT32> guiDistribution(0, UINT32_MAX);
+
+// Pregenerated pseudo-random numbers.
 UINT32 guiPreRandomIndex = 0;
 UINT32 guiPreRandomNums[ MAX_PREGENERATED_NUMS ];
 
+/// Pre-generated pseudo-random number engine.
+struct PreRandomEngine {
+	typedef UINT32 result_type;
```

Comment:
I'd remove this typedef, since it doesn't add much (pretty obvious the types should match) and max() uses a fixed constant anyway.

Reply:
Can't be removed, `std::uniform_int_distribution` requires that typedef and the other member functions.